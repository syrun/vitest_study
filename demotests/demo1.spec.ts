import { describe, expect, it } from 'vitest'

function toUpperCase(str: string) {
  return str.toUpperCase() + '1'
}

describe('matchSnapshot', () => {
  it('toUpperCase', () => {
    const result = toUpperCase('foo')
    // 和之前缓存的快照进行对比，如果一样则测试通过
    // 如果不一样
    // 要么是函数输出被意外改动
    // 要么是快照版本需要更新（所有bug已被改完）
    expect(result).toMatchSnapshot()
  })
})