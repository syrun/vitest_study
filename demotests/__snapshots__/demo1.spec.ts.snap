// Vitest Snapshot v1

exports[`matchSnapshot > toUpperCase 1`] = `"FOO1"`;
